ACadSharp/IO/Templates/CadSplineTemplate.cs

```diff
@@ -0,0 +1,105 @@
+﻿using ACadSharp.Entities;
+using System;
+using System.Collections.Generic;
+
+namespace ACadSharp.IO.Templates
+{
+	internal class CadSplineTemplate : CadEntityTemplate
+	{
+        private CSMath.XYZ _currentControlPoint = new CSMath.XYZ();
+        private CSMath.XYZ _currentFitPoint = new CSMath.XYZ();
+
+        public CadSplineTemplate(Spline entity) : base(entity) { }
+
+        public override bool CheckDxfCode(int dxfcode, object value)
+        {
+            bool found = base.CheckDxfCode(dxfcode, value);
+            if (found)
+                return found;
+
+            var spline = this.CadObject as Spline;
+
+            switch(dxfcode)
+            {
+                //--- Knot
+                case 40:
+                    spline.Knots.Add((double)value);
+                    return true;
+
+                //--- Weight
+                case 41:
+                    spline.Weights.Add((double)value);
+                    return true;
+
+                //--- ControlPoint
+                case 10:
+                    this._currentControlPoint = new CSMath.XYZ(
+                        (double)value,
+                        _currentControlPoint.Y,
+                        _currentControlPoint.Z
```

Comment:
PR looks really good, but in this point I've spotted a reading error:
When this code is read, this should reset the field ```_currentControlPoint``` like ```new CSMath.XYZ((double)value, 0, 0)``` the way is written right now, at the second point it will read the Y and Z from the first one.

Reply:
Thank you for correction, I do check the proposed changes.